File: tests/fixtures/java_project/src/main/java/com/example/MemoryStorage.java
package com.example;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * In-memory storage implementation.
 * Stores users in a HashMap for fast access.
 */
public class MemoryStorage extends AbstractStorage {
    private Map<String, User> users = new HashMap<>();

    /**
     * Creates a new in-memory storage.
     */
    public MemoryStorage() {
        super("memory");
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void save(User user) {
        users.put(user.getEmail(), user);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public User load(String email) {
        return users.get(email);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public boolean delete(String email) {
        return users.remove(email) != null;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public List<User> list() {
        return new ArrayList<>(users.values());
    }

    /**
     * Returns the number of stored users.
     *
     * @return The user count
     */
    public int size() {
        return users.size();
    }

    /**
     * Clears all stored users.
     */
    public void clear() {
        users.clear();
    }
}

File: test/fixtures/php_project/src/MemoryStorage.php
<?php

declare(strict_types=1);

namespace LspcmdFixture;

/**
 * Stores users in memory.
 */
class MemoryStorage implements Storage
{
    /** @var array<string, User> */
    private array $users = [];

    /**
     * Saves a user to memory.
     *
     * @param User $user The user to save
     */
    public function save(User $user): void
    {
        $this->users[$user->getEmail()] = $user;
    }

    /**
     * Loads a user by email.
     *
     * @param string $email The email to search for
     * @return User|null The user if found
     */
    public function load(string $email): ?User
    {
        return $this->users[$email] ?? null;
    }

    /**
     * Deletes a user by email.
     *
     * @param string $email The email of the user to delete
     * @return bool True if the user was deleted
     */
    public function delete(string $email): bool
    {
        if (isset($this->users[$email])) {
            unset($this->users[$email]);
            return true;
        }
        return false;
    }

    /**
     * Lists all users.
     *
     * @return User[] All users in the storage
     */
    public function list(): array
    {
        return array_values($this->users);
    }
}
